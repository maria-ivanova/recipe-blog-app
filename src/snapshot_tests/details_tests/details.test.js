import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import Routing from '../helpers/router.js';
import Details from '../../components/Details.js';

import { library } from '@fortawesome/fontawesome-svg-core';
import { faHeart, faEdit } from '@fortawesome/free-solid-svg-icons';
library.add(faHeart, faEdit);

describe('Details Component', () => {
    it('should render page layout with auth user', () => {
        const tree = renderer.create(
            <Routing value={{ displayName: "test", email: "test@test.com", uid: "1" }}>
                <Details match={{ params: { id: '1' } }}/>
            </Routing>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })

    it('should render page layout with non-auth user', () => {
        const tree = renderer.create(
            <Routing >
                <Details match={{ params: { id: '1' } }}/>
            </Routing>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })

    it('should render button edit', async () => {
        const wrapper = mount(shallow(
            <Routing value={{ displayName: "test", email: "test@test.com", uid: "1" }}>
                <Details match={{ params: { id: '1' } }}/>
            </Routing>
        ).get(0))

        let component = wrapper.find('Details');
        component.setState({ creatorId: '1' });
        component = component.update();

        expect(component.find('a').text()).toEqual(' Редактирай');
      });

      it('should render button like', async () => {
        const wrapper = mount(shallow(
            <Routing value={{ displayName: "test", email: "test@test.com", uid: "1" }}>
                <Details match={{ params: { id: '1' } }}/>
            </Routing>
        ).get(0))

        let component = wrapper.find('Details');
        component.setState({ likesArr: [2] });
        component = component.update();

        expect(component.find('button').text()).toEqual('Добави в любими');
      });

      it('should render is liked', async () => {
        const wrapper = mount(shallow(
            <Routing value={{ displayName: "test", email: "test@test.com", uid: "1" }}>
                <Details match={{ params: { id: '1' } }}/>
            </Routing>
        ).get(0))

        let component = wrapper.find('Details');
        component.setState({ likesArr: ['1'] });
        component = component.update();

        expect(component.find('.favorites_info').text()).toEqual(' Добавена в любими');
      });
})