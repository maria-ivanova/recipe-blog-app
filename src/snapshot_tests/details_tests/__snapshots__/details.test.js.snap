// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component should render page layout with auth user 1`] = `
Array [
  <div
    className="notifications 
                             null 
                             "
  >
    <span>
      
    </span>
  </div>,
  <section
    className="sec"
  >
    <div
      className="banner_item"
    >
      <h2
        className="title"
      >
        
      </h2>
    </div>
  </section>,
  <section
    className="sec content_sec"
  >
    <div
      className="container"
    >
      <div
        className="item_layout"
      >
        <div
          className="left_part"
        >
          <div
            className="img_box"
          >
            <img
              alt="recipe pic"
              src=""
            />
          </div>
          <div
            className="features"
          >
            <div
              className="features_item"
            >
              <span
                className="icon"
              >
                <img
                  alt="book"
                  src="i_book.svg"
                />
              </span>
              <div
                className="info"
              >
                <span
                  className="title"
                >
                  Категория
                </span>
                <br />
                <span>
                  
                </span>
              </div>
            </div>
            <div
              className="features_item"
            >
              <span
                className="icon"
              >
                <img
                  alt="alarm"
                  src="i_alarm.svg"
                />
              </span>
              <div
                className="info"
              >
                <span
                  className="title"
                >
                  Време за приготвяне
                </span>
                <br />
                <span>
                  
                </span>
              </div>
            </div>
            <div
              className="features_item"
            >
              <span
                className="icon"
              >
                <img
                  alt="dish"
                  src="i_dish.svg"
                />
              </span>
              <div
                className="info"
              >
                <span
                  className="title"
                >
                  Порции
                </span>
                <br />
                <span>
                  
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="right_part"
        >
          <h2
            className="page_title"
          >
            
          </h2>
          <h3
            className="inner_title"
          >
            Необходими съставки:
          </h3>
          <div
            className="row"
          >
            
          </div>
        </div>
        <div
          className="description"
        >
          <h3
            className="inner_title"
          >
            Начин на приготвяне:
          </h3>
          <div
            className="row"
          >
            
          </div>
        </div>
        <div
          className="item_meta"
        >
          <div
            className="meta_info"
          >
            Публикувано от 
            <span>
              
            </span>
             на  
            <span>
              1970 M01 01
            </span>
          </div>
          <button
            className="btn btn"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-heart fa-w-16 fa"
              data-icon="heart"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Добави в любими
          </button>
        </div>
      </div>
    </div>
  </section>,
]
`;

exports[`Details Component should render page layout with non-auth user 1`] = `
Array [
  <div
    className="notifications 
                             null 
                             "
  >
    <span>
      
    </span>
  </div>,
  <section
    className="sec"
  >
    <div
      className="banner_item"
    >
      <h2
        className="title"
      >
        
      </h2>
    </div>
  </section>,
  <section
    className="sec content_sec"
  >
    <div
      className="container"
    >
      <div
        className="item_layout"
      >
        <div
          className="left_part"
        >
          <div
            className="img_box"
          >
            <img
              alt="recipe pic"
              src=""
            />
          </div>
          <div
            className="features"
          >
            <div
              className="features_item"
            >
              <span
                className="icon"
              >
                <img
                  alt="book"
                  src="i_book.svg"
                />
              </span>
              <div
                className="info"
              >
                <span
                  className="title"
                >
                  Категория
                </span>
                <br />
                <span>
                  
                </span>
              </div>
            </div>
            <div
              className="features_item"
            >
              <span
                className="icon"
              >
                <img
                  alt="alarm"
                  src="i_alarm.svg"
                />
              </span>
              <div
                className="info"
              >
                <span
                  className="title"
                >
                  Време за приготвяне
                </span>
                <br />
                <span>
                  
                </span>
              </div>
            </div>
            <div
              className="features_item"
            >
              <span
                className="icon"
              >
                <img
                  alt="dish"
                  src="i_dish.svg"
                />
              </span>
              <div
                className="info"
              >
                <span
                  className="title"
                >
                  Порции
                </span>
                <br />
                <span>
                  
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="right_part"
        >
          <h2
            className="page_title"
          >
            
          </h2>
          <h3
            className="inner_title"
          >
            Необходими съставки:
          </h3>
          <div
            className="row"
          >
            
          </div>
        </div>
        <div
          className="description"
        >
          <h3
            className="inner_title"
          >
            Начин на приготвяне:
          </h3>
          <div
            className="row"
          >
            
          </div>
        </div>
        <div
          className="item_meta"
        >
          <div
            className="meta_info"
          >
            Публикувано от 
            <span>
              
            </span>
             на  
            <span>
              1970 M01 01
            </span>
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
